enclave/core/sgx/enter.S

```diff
@@ -65,16 +69,28 @@ oe_enter:
 
 .determine_entry_type:
     // Check if this is exception dispatching request
-    // Abort on the eenter if cssa greater than one, which
+    // Return on the eenter if cssa greater than one, which
     // should not occur because OE assumes the enclave with nssa=2
     cmp $1, %rax
     je .exception_entry
-    ja .abort
+    ja .return
```

Comment:
If two different signals are send rapidly, one after the other, then the second signal could trigger this return and go unprocessed.

Reply:
There two cases:
1. Signal during the first-stage exception handling (cssa == 1). This signal will abort the enclave given that we configure the enclave with nssa = 2. That is, if the cssa == 2, the SGX hardware will return #GP. To avoid such failure, our host library blocks one of the signals in this case: https://github.com/openenclave/openenclave/blob/cf9ec0c99ee64c1344d748ef462f91847770d4b5/host/sgx/exception.c#L33

2. Signal during the second-stage exception handling. Current design explicitly block the such signals (if it's sent by user, not the hard fails such as SIGSEGV).  The idea is to preventing the nested interrupts. Mystikos, for example, queues up the signal upon sending so that it can be handled later.